index.js

```diff
@@ -551,6 +552,19 @@ Browserify.prototype._createDeps = function (opts) {
                 return through();
             }
         }
+
+        if (opts.commondir === false && opts.builtins === false) {
+          opts.insertGlobalVars = xtend({
+            __dirname: function(file, basedir) {
+              var dir = path.dirname(path.relative(basedir, file));
+              return 'require("path").join(__dirname,"' + dir.split(path.sep).join('","') + '")';
+            },
+            __filename: function(file) {
+              var filename = path.relative(basedir, file);
```

Comment:
Possibly-stupid first question:
How come `__dirname` has the parameters `(file, basedir)`, but `__filename` only has the parameter `file`? In that case, where does the `basedir` variable in this line come from?